clients/osv/src/main/kotlin/OsvService.kt

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2022 EPAM Systems, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.clients.osv
+
+import java.io.IOException
+import java.util.concurrent.ConcurrentLinkedQueue
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.atomic.AtomicReference
+
+import okhttp3.OkHttpClient
+
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+/**
+ * This class wraps the OSV Rest API client in order to make its use simpler and less error-prone.
+ */
+class OsvService(val serverUrl: String = SERVER_URL_DEFAULT_VALUE, httpClient: OkHttpClient? = null) {
+    companion object {
+        const val SERVER_URL_DEFAULT_VALUE = OsvApiClient.SERVER_URL_PRODUCTION
+    }
+
+    private val client = OsvApiClient.create(serverUrl, httpClient)
+
+    /**
+     * Get the vulnerabilities for the package matching the given [request].
+     */
+    fun getVulnerabilitiesForPackage(
+        request: VulnerabilitiesForPackageRequest
+    ): Result<List<Vulnerability>> {
+        val response = client.getVulnerabilitiesForPackage(request).execute()
```

Comment:
BTW, in this line it occurred to me that it probably would have been nicer if you had implemented the API by making use of [Retrofit's suspend functions](https://proandroiddev.com/suspend-what-youre-doing-retrofit-has-now-coroutines-support-c65bd09ba067#1cb3), also see the other client implementations.

Reply:
Can you explain in what way that would have been nicer?